Source/SocketStringReader.swift

```diff
@@ -40,33 +40,34 @@ struct SocketStringReader {
     
     @discardableResult
     mutating func advance(by: Int) -> String.Index {
-        currentIndex = message.index(currentIndex, offsetBy: by)
+        currentIndex = message.characters.index(currentIndex, offsetBy: by)
         
         return currentIndex
     }
     
     mutating func read(count: Int) -> String {
-        let readString = message[currentIndex..<message.index(currentIndex, offsetBy: count)]
+        let readString = message[currentIndex..<message.characters.index(currentIndex, offsetBy: count)]
+        
         advance(by: count)
         
         return String(readString)
     }
     
     mutating func readUntilOccurence(of string: String) -> String {
-        let substring = message.utf16[currentIndex...]
```

Comment:
Have you verified this with unicode characters and socket.io 1.0?

Reply:
Think I got these changes from your update to Swift 4.0 but I reverted them and just updated the Swift version and NSData -> Data change.